src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -267,6 +267,22 @@ private void HandleChangesForAutoIndexes(DatabaseRecord record, long index, List
             {
                 _documentDatabase.DatabaseShutdown.ThrowIfCancellationRequested();
 
+                // This will handle the case of indexes that are been migrated from v5.3 and older databases.
+                // Since the id does not exist in those cases we will assign to each field a new value starting
+                // from 1 in order to do not collide with the 0 which is reserved for the document id. 
+                var i = 0;
```

Comment:
@maciejaszyk We found this case that only triggers when we import indexes from older versions. Can you check if this is correct and we are not missing any particular assignation of field ids?

Reply:
Can you provide some test data for it?

Reply:
Yeah, concrete sample index definition would be great to verify that. Assigning the field ids is handled at different level and we shouldn't need this code here (in particular it was added only to `HandleChangesForAutoIndexes()` we also have `HandleChangesForStaticndexes()`). Exported index definition itself doesn't contain the field ids.

Reply:
The way it works is that you can take sample data export from 5.3 instance with auto indexes and import that at 5.4

Because we added the  `Id` field, it is set to `0` in the `DatabaseRecord`, which is wrong.
This is a quickfix to avoid this.

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-18864